checks/evaluation/vulnerabilities_test.go

```diff
@@ -17,62 +17,119 @@ package evaluation
 import (
 	"testing"
 
-	"github.com/ossf/scorecard/v4/checker"
-	"github.com/ossf/scorecard/v4/clients"
+	sce "github.com/ossf/scorecard/v4/errors"
+	"github.com/ossf/scorecard/v4/finding"
 	scut "github.com/ossf/scorecard/v4/utests"
 )
 
 // TestVulnerabilities tests the vulnerabilities checker.
 func TestVulnerabilities(t *testing.T) {
 	t.Parallel()
 	//nolint
-	type args struct {
-		name string
-		r    *checker.VulnerabilitiesData
-	}
 	tests := []struct {
 		name     string
-		args     args
-		want     checker.CheckResult
+		findings     []finding.Finding
+		result     scut.TestReturn
 		expected []struct {
 			lineNumber uint
 		}
 	}{
 		{
 			name: "no vulnerabilities",
-			args: args{
-				name: "vulnerabilities_test.go",
-				r: &checker.VulnerabilitiesData{
-					Vulnerabilities: []clients.Vulnerability{},
+			findings: []finding.Finding {
+				{
+					Probe:   "hasOSVVulnerabilities",
+					Outcome: finding.OutcomePositive,
 				},
 			},
-			want: checker.CheckResult{
+			result: scut.TestReturn{
 				Score: 10,
 			},
 		},
 		{
-			name: "one vulnerability",
-			args: args{
-				name: "vulnerabilities_test.go",
-				r: &checker.VulnerabilitiesData{
-					Vulnerabilities: []clients.Vulnerability{
-						{
-							ID: "CVE-2019-1234",
-						},
-					},
+			name: "three vulnerabilities",
+			findings: []finding.Finding {
+				{
+					Probe:   "hasOSVVulnerabilities",
+					Outcome: finding.OutcomeNegative,
+				},
+				{
+					Probe:   "hasOSVVulnerabilities",
+					Outcome: finding.OutcomeNegative,
+				},
+				{
+					Probe:   "hasOSVVulnerabilities",
+					Outcome: finding.OutcomeNegative,
```

Comment:
this is one where the integer value could come into play, but if we start using the source info Rex mentioned, then perhaps keeping it one finding per vuln would be ideal.

Reply:
1 negative finding per vuln group/alias.

Differentiates between 10 vs 100 vulns. Can filter out CVEs you don't care about. Can point to specific lines in the code scanning dashboard.

Reply:
I believe the current probe does that ([ref](https://github.com/ossf/scorecard/pull/3487/files#diff-16338b5a380d24d093b88363f732b15a6407fa51d2a5db074d9a428ca9a6bf1cR22)), so closing.